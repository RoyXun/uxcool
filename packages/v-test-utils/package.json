{
  "name": "@suning/v-test-utils",
  "version": "1.0.10",
  "main": "src/index.js",
  "module": "src/index.js",
  "license": "MIT",
  "scripts": {
    "start": "npm run server",
    "compile": "v-tools run compile",
    "server": "v-tools run server"
  },
  "peerDependencies": {
    "vue": ">= 2.5.*"
  },
  "devDependencies": {
    "@suning/v-tools": "^1.0.35",
    "@vue/test-utils": "^1.0.0-beta.16"
  },
  "dependencies": {
    "@suning/v-trigger": "^1.0.64",
    "@vue/server-test-utils": "^1.0.0-beta.30",
    "jquery": "^3.3.1",
    "vue-server-renderer": "^2.6.11"
  }
}
