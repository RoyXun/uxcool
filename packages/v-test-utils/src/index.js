import $ from 'jquery';

export * from '@vue/test-utils';

export { $, $ as jQuery };
// ms
export async function waitTime(delay) {
  await new Promise((resolve) => {
    setTimeout(() => {
      resolve();
    }, delay);
  });
}
