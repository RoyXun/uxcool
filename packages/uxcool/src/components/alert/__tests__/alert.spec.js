import { mountPickerFactory, waitTime, triggerEvent } from '@cloud-sn/v-test-utils';
import Alert from '../index';

describe('alert', () => {
  const mountAlert = mountPickerFactory(Alert);
  it('render banner alert correctly', async () => {
    const wrapper = await mountAlert({
      propsData: {
        banner: true,
      },
    });
    expect(wrapper.contains('.ux-alert-banner')).toBeTruthy();
  });

  it('render show icon correctly', async () => {
    const wrapper = await mountAlert({
      propsData: {
        showIcon: true,
      },
    });
    expect(wrapper.contains('.ux-alert-icon')).toBeTruthy();
    wrapper.setProps({ showIcon: false });
    await waitTime();
    expect(wrapper.contains('.ux-alert-icon')).not.toBeTruthy();
  });

  it('render closable and close correctly', async () => {
    const closeFn = jest.fn();
    const afterCloseFn = jest.fn();
    const wrapper = await mountAlert({
      propsData: {
        closable: false,
      },
      listeners: {
        close: closeFn,
        'after-close': afterCloseFn,
      },
    });
    expect(wrapper.contains('.ux-alert-close-icon')).not.toBeTruthy();
    wrapper.setProps({ closable: true });
    await waitTime();
    expect(wrapper.contains('.ux-alert-close-icon')).toBeTruthy();

    const closeBtn = wrapper.find('.ux-alert-close-icon');
    await triggerEvent(closeBtn, 'click');
    expect(closeFn).toHaveBeenCalledTimes(1);
    expect(wrapper.contains('.ux-alert')).not.toBeTruthy();

    wrapper.vm.onAfterClose();
    expect(afterCloseFn).toHaveBeenCalledTimes(1);
  });

  it('render with props or slots correctly', async () => {
    const closeText = 'close';
    const message = 'this is message';
    const description = 'this is desc';
    const propsWrapper = await mountAlert({
      propsData: {
        closeText,
        message,
        description,
      },
    });
    expect(propsWrapper.html()).toMatchSnapshot();
    const slotsWrapper = await mountAlert({
      slots: {
        closeText,
        message,
        description,
      },
    });
    expect(slotsWrapper.html()).toMatchSnapshot();
  });

  it('render default icon or custom icon correctly', async () => {
    const wrapper = await mountAlert({
      propsData: {
        type: 'warning',
        showIcon: true,
      },
    });
    expect(wrapper.contains('.fu-exclamation_circle')).toBeTruthy();

    wrapper.setProps({ iconType: 'happy-d' });
    await waitTime();
    expect(wrapper.contains('.fu-happy-d')).toBeTruthy();
  });
});
