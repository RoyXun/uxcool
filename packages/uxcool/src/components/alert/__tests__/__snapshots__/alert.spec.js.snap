// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alert render with props or slots correctly 1`] = `<div class="ux-alert ux-alert-info ux-alert-no-icon ux-alert-with-description"><span class="ux-alert-message">this is message</span><span class="ux-alert-description">this is desc</span><a class="ux-alert-close-icon">close</a></div>`;

exports[`alert render with props or slots correctly 2`] = `<div class="ux-alert ux-alert-info ux-alert-no-icon ux-alert-with-description"><span class="ux-alert-message">this is message</span><span class="ux-alert-description">this is desc</span><a class="ux-alert-close-icon">close</a></div>`;
